src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -20,7 +17,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     /// </summary>
     internal sealed class TupleTypeSymbol : NamedTypeSymbol, ITupleTypeSymbol
     {
-        private readonly NamedTypeSymbol _underlyingType;
+        private readonly OneOrMany<NamedTypeSymbol> _underlyingTypeChain;
```

Comment:
I agree, this is an overkill, we don't need this chain because the very first (top-level) underlying tuple type has all the information already. Having the chain doesn't simplify anything, it uses more memory and requires more maintenance like with generic substitution. With generic substitution, for example we are going to substitute the same types multiple time, the longer the chain, more duplication will be there. Given a chain of length N, the last type in the chain will be substituted N times, the one before that (N-1) times, etc. Whereas, all we need to substitute is the first (top-level) type, then all types in the chain will be substituted only once.